src/torch_constrained/constrained_optimizer.py

```diff
@@ -1,359 +1,289 @@
-import functools
 import inspect
 import warnings
-from typing import Type, Callable, Optional
+from functools import partial
+from typing import Type, List, Callable, Optional, Dict
 
 import torch
-import torch.nn
-
+from .multipliers import _SparseMultiplier, _DenseMultiplier
 
 class ConstrainedOptimizer(torch.optim.Optimizer):
     def __init__(
             self,
-            loss_optimizer: Type[torch.optim.Optimizer],
-            constraint_optimizer: Type[torch.optim.Optimizer],
-            lr_x,
-            lr_y,
+            primal_optim_class: Type[torch.optim.Optimizer],
+            primal_kwargs: Dict,
             primal_parameters,
-            augmented_lagrangian_coefficient=False,
+            dual_optim_class: Optional[Type[torch.optim.Optimizer]]=None,
+            dual_kwargs: Optional[Dict]=None,
+            ineq_init: Optional[List[torch.Tensor]]=None,
+            eq_init: Optional[List[torch.Tensor]]=None,
+            augmented_lagrangian_coeff=False,
             alternating=False,
-            shrinkage: Optional[Callable] = None,
-            dual_dtype=None,
-            primal_kwargs={},
-            dual_kwargs={},
-    ):
+            shrinkage: Optional[Callable]=None,
+            dual_reset=False,
+            verbose=False,
+        ):
+        
+        # Instantiate user-specified primal optimizer
         if inspect.isgenerator(primal_parameters):
             primal_parameters = list(primal_parameters)
-
-        self.primal_optimizer = loss_optimizer(primal_parameters, lr_x, **primal_kwargs)
-        self.dual_optimizer_class = functools.partial(constraint_optimizer, lr=lr_y, **dual_kwargs)
-
-        self.dual_optimizer = None
-        self.augmented_lagrangian_coefficient = augmented_lagrangian_coefficient
-        self.alternating = alternating
-        self.shrinkage = shrinkage
-        self.dual_dtype = dual_dtype
+        self.primal_optimizer = primal_optim_class(primal_parameters, **primal_kwargs)
+        self.primal_parameters = primal_parameters
+
+        # Initialization values for the dual variables
+        self.dual_init_done = False
+        self.ineq_init = ineq_init
+        self.eq_init = eq_init
+
+        # Create flag for execution in un-constrained setting
+        self.is_constrained = dual_optim_class is not None 
```

Comment:
the problem could be constrained, but without a dual optimizer (regularized), `self.dual_optimizer is None` or something similar should be enough.